src/org/daisy/dotify/api/formatter/NumeralStyle.java

```diff
@@ -1,72 +1,113 @@
-	 * @param i
-	 *            the number
-	 * @return returns the formatted number
-	 */
-	public String format(int i) {
-		switch (this) {
-			case ROMAN:
-			case UPPER_ROMAN:
-				return RomanNumeral.int2roman(i);
-			case LOWER_ROMAN:
-				return RomanNumeral.int2roman(i).toLowerCase();
-			case ALPHA:
-			case UPPER_ALPHA:
-				return AlphaNumeral.int2alpha(i);
-			case LOWER_ALPHA:
-				return AlphaNumeral.int2alpha(i).toLowerCase();
-			case DECIMAL_LEADING_ZERO:
-				return (i<10?"0":"")+i;
-			case DEFAULT: 
-			case DECIMAL:
-			default:
-				return "" + i;
-		}
-	}
-};
\ No newline at end of file
+    /**
+     * Defines default numeral style, alias of DECIMAL
+     */
+    DEFAULT,
+    /**
+     * Defines a decimal numeral style
+     */
+    DECIMAL,
+    /**
+     * Defines a number with a single leading zero if the number is &lt; 10 (01, 02, 03, etc.)
+     */
+    DECIMAL_LEADING_ZERO,
+    /**
+     * Defines roman numeral style, alias of UPPER_ROMAN
+     */
+    ROMAN,
+    /**
+     * Defines upper roman numeral style
+     */
+    UPPER_ROMAN,
+    /**
+     * Defines lower roman numeral style
+     */
+    LOWER_ROMAN,
+    /**
+     * Defines alpha numeral style, alias of UPPER_ALPHA
+     */
+    ALPHA,
+    /**
+     * Defines upper alpha numeral style
+     */
+    UPPER_ALPHA,
+    /**
+     * Defines lower alpha numeral style
+     */
+    LOWER_ALPHA;
+    
+    /**
+     * Formats the numeral with the given style
+     * 
+     * @param i the number
+     * @return returns the formatted number
+     */
+    public String format(int i) {
+        switch (this) {
+            case ROMAN:
+            case UPPER_ROMAN:
+                return RomanNumeral.int2roman(i);
+            case LOWER_ROMAN:
+                return RomanNumeral.int2roman(i).toLowerCase();
+            case ALPHA:
+            case UPPER_ALPHA:
+                return AlphaNumeral.int2alpha(i);
+            case LOWER_ALPHA:
+                return AlphaNumeral.int2alpha(i).toLowerCase();
+            case DECIMAL_LEADING_ZERO:
+                return (i<10?"0":"")+i;
+            case DEFAULT: 
+            case DECIMAL:
+            default:
+                return "" + i;
+        }
+    }
+    
+    /**
+     * Formats the string with the given style
+     * 
+     * If the formatting does not succeed, no exception is thrown, but
+     * a warning will be logged and the input will be returned
+     * 
+     * @param s the string
+     * @return returns the formatted string
+     */
+    public String format(String s) {
+        if (this == DEFAULT) {
+            return s;
+        }
+        Integer i = toInt(s);
+        if (i == null) {
+            Logger.getLogger(this.getClass().getCanonicalName()).warning("Failed to convert this string to an integer: " + s);
+            return s;
+        }
+        return format(i);
+    }
+    
+    /**
+     * Converts a string to an Integer
+     * Rounds double values to the nearest integer
```

Comment:
Wouldn't it be more logical to issue the "Failed to convert this string to an integer" warning if the number has a decimal point? Asking to format "1.0" or "1.1" to a roman number doesn't make sense.